ibc-apps/ics721-nft-transfer/src/handler/mod.rs

```diff
@@ -21,30 +21,35 @@ pub fn refund_packet_nft_execute(
         .sender
         .clone()
         .try_into()
-        .map_err(|_| NftTransferError::ParseAccountFailure)?;
+        .map_err(|_| NftTransferError::FailedToParseAccount)?;
 
     if is_sender_chain_source(
         packet.port_id_on_a.clone(),
         packet.chan_id_on_a.clone(),
         &data.class_id,
     ) {
         data.token_ids.as_ref().iter().try_for_each(|token_id| {
-            ctx_a.unescrow_nft_execute(
-                &sender,
-                &packet.port_id_on_a,
-                &packet.chan_id_on_a,
-                &data.class_id,
-                token_id,
-            )
+            ctx_a
```

Comment:
it seems that the errors are not propagated. `try_for_each` just collects the errors.

Reply:
The `try_for_each` actually propagates the error on failures.

Reply:
Ah. That line break before `else` threw me off.  You are right, it correctly returns the result from `try_for_each`.